src/core/midi/MidiApple.cpp

```diff
@@ -615,7 +615,11 @@ char * MidiApple::getFullName(MIDIEndpointRef &endpoint_ref)
 	char * deviceName = getName(device);
 	char * endPointName = getName(endpoint_ref);
 	qDebug("device name='%s' endpoint name='%s'",deviceName,endPointName);
-	char * fullName = (char *)malloc(strlen(deviceName) + strlen(":") + strlen(endPointName)+1);
+	size_t deviceName_len = 0;
+	if(deviceName != nil) deviceName_len = strlen(deviceName);
+	size_t endPointName_len = 0;
+	if(endPointName != nil) endPointName_len = strlen(endPointName);
+	char * fullName = (char *)malloc(deviceName_len + endPointName_len + 2);
```

Comment:
I'm not very familiar with this part of the code, but it looks good to me except for some code style issues (camelCase naming and pointer declaration). I think it could be shortened using ternary expressions too:

```
size_t deviceNameLen = deviceName == nullptr ? 0 : strlen(deviceName);
size_t endPointNameLen = endPointName == nullptr ? 0 : strlen(endPointName);
char* fullName = (char *)malloc(deviceNameLen + endPointNameLen + 2)
```

I also changed `nil` to `nullptr` because I think it's what should be used there. On that last line I'll wait on someone else's opinion whether obscuring the `strlen(":")` is fine, but bottom line it does the same thing so maybe just adding a comment there explaining the `+2` would be fine.

Reply:
getName uses `nil` and not `nullptr`. Should that be updated as well?

https://github.com/LMMS/lmms/blob/f06f46fc2fec771607f5ebe282d19270c3b60f45/src/core/midi/MidiApple.cpp#L401-L413

Reply:
Sure! Some of the older code might have code style issues or little issues, feel free to update it as well :)
If you end up fixing too much code that is unrelated to the main issue of the PR, then maybe it's better to submit those as a new refactoring PR for easier review though

Reply:
So, fun facts, at least with gcc both the if and tertiary format compile to near the same assembly. The `if` version is at 11 lines, and the `?:` version is 12.

Reply:
I'm surprised that there's any difference at all.

Reply:
A bit ironic the shorter code compiles to a slightly longer ASM :laughing: